packages/core/src/domain/internalMonitoring/internalMonitoring.ts

```diff
@@ -31,18 +39,34 @@ const monitoringConfiguration: {
   sentMessageCount: number
 } = { maxMessagesPerPage: 0, sentMessageCount: 0 }
 
-let onInternalMonitoringMessageCollected: ((message: MonitoringMessage) => void) | undefined
+let monitoringMessageObservable: Observable<MonitoringMessage> | undefined
```

Comment:
FMU you are proposing to do:
```ts
function addToMonitoring(message: MonitoringMessage) {
  if (
    monitoringConfiguration.sentMessageCount < monitoringConfiguration.maxMessagesPerPage
  ) {
    monitoringConfiguration.sentMessageCount += 1
    getMonitoringMessageObservable().notify(message)
  }
}
```
I found it clearer to explicitly create the observable on the start methods and since we have a reset method, I would find it surprising that after a reset, adding a message recreate the observable.
wdyt?

Reply:
the design seems different to me:
- for xhr/fetch, we expose an init function that get or create the singleton
- for replayStats, we only expose methods to interact with the replay stats, none to explicitly initialize it

here we expose a start method, so it feels weird to me to create it outside of it. 

However, for replay stats the undefined case is also handled in [getReplayStats](https://github.com/DataDog/browser-sdk/blob/50607ccb3e1408a46f8e99f1bc3987029e97d4e4/packages/rum/src/domain/replayStats.ts#L23) with a `?`. 
Would you prefer this approach?

Reply:
My point is, appart from not being able to treeshake it, there is no real reason not to do:
```ts
let monitoringMessageObservable = new Observable<MonitoringMessage>()
```
where `monitoringMessageObservable` is never undefined.

If you think that a getter to work around the treeshakability limitation is not worth it, then let's keep your solution as it is.